### DISCUSSION TITLE
How does a server proactively send messages to a client?

### DISCUSSION THREAD

**Discussion by qilivip** [actix/examples#548] [Q&A]
https://github.com/actix/examples/tree/master/websockets/chat-broker
How does a server proactively send messages to a client?
Thanks！

**Comment by jakswa** [comment]
If you only want server push, I would read up on:
- https://github.com/actix/examples/tree/master/server-sent-events
- https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events

Combined with HTTP/2 connection reuse, I think (but may be wrong to think that) it is very scalable to "pull" with new requests, and get pushed payloads over SSE.
